tests/core/test_http.py

```diff
@@ -0,0 +1,33 @@
+from typing import Dict
+
+import pytest
+
+from great_expectations.core.http import DEFAULT_TIMEOUT, session_factory
+
+
+@pytest.mark.unit
+def test_session_factory_contains_appropriate_headers(
+    ge_cloud_access_token: str, request_headers: Dict[str, str]
+) -> None:
+    session = session_factory(access_token=ge_cloud_access_token)
+    for key, val in request_headers.items():
```

Comment:
Do you want to test for equality or are you expecting there might be more headers in the `session.headers`?

Reply:
there are more headers within the session by default - can go either way here but opted to keep the defaults in for now

Reply:
sounds good.